tests/typ/lint/scripted-warnings.typ

```diff
@@ -0,0 +1,69 @@
+/// Test scripted warnings (Inspired by https://github.com/typst/typst/issues/1276#issuecomment-1560091418).
+// Ref: false
+
+---
+#set heading(numbering: "1.")
+
+#let myref(label) = locate(loc => {
+    if query(label,loc).len() != 0 {
+        ref(label)
+    } else {
+// Warning: 14-61 could not find reference <test>
+        warn("could not find reference <" + str(label) + ">")
+    }
+})
+
+= Second <test2>
+
+#myref(<test>)
+
+---
+// This test verifies warn calls are not invoked if the user defined function calling it is not invoked 
+#set heading(numbering: "1.")
+
+#let myref(label) = locate(loc => {
+    if query(label,loc).len() != 0 {

```

Comment:
indent is always 2 spaces

Reply:
I think it's indented by 2 in the typ files I edited now!